### ISSUE TITLE
Windows guest CPU hot-remove

### ISSUE THREAD

**Issue by weltling** [cloud-hypervisor/cloud-hypervisor#2457]
This is a follow up from #1799. A closer look is to be taken at the CPU remove functionality for the Windows guests.

**Comment by sboeuf** [comment]
Have you tried latest master? I'm asking because we merged #2455 today and this might be the reason why you're having some trouble unplugging vCPUs.

**Comment by weltling** [comment]
After rebasing to the latest no improvement is to see. I'm going to make another round of debugging and compare with QEMU again.

Btw the latest QEMU pulled in the distro i use also made the hotplug issue work with QEMU. It was not the case previously. I guess it might be expedient to check the changes in the latest QEMU, too.

Thanks

**Comment by weltling** [comment]
After some research I've got now confirmed, that Windows doesn't support CPU hot-remove.

An additional check I went for was testing again with the latest version of QEMU. The steps as described before worked now, however not the removal. Furthermore, a little difference is that QEMU also doesn't remove a CPU after `device_del` and `device_del` doesn't fail (same way it doesn't fail as discussed previously on the memory hotplug issue) and the CPU is also present after reboot. CH however will eject the CPU and Windows will see the correct number of CPUs after the reboot.

With this - I'm not sure further tests would make sense. Perhaps could add a test to see the number of CPUs is adapted after the reboot. And to add the documentation.

Thanks
